src/org/opendatakit/briefcase/ui/pull/PullPanel.java

```diff
@@ -63,11 +63,15 @@ public PullPanel(TransferPanelForm view, TransferForms forms, BriefcasePreferenc
     getContainer().addComponentListener(analytics.buildComponentListener("Pull"));
 
     // Read prefs and load saved remote server if available
-    this.source = RemoteServer.readPreferences(tabPreferences).flatMap(view::preloadSource);
-    this.source.ifPresent(source -> {
-      forms.load(source.getFormList());
-      view.refresh();
-      updateActionButtons();
+    source = RemoteServer.readPreferences(tabPreferences).flatMap(view::preloadSource);
+    source.ifPresent(source -> {
```

Comment:
So, following my rationale from my first comment, what we should avoid is having something in `source` other than an `Optional.empty()` when the stored conf values won't produce a successful connection to the remote Aggregate server.